superset-frontend/src/components/Select/SupersetStyledSelect.tsx

```diff
@@ -259,36 +257,43 @@ function styled<
     };
 
     const theme = useTheme();
+    const menuPortalTargetRef = React.useRef() as React.MutableRefObject<HTMLDivElement>;
 
     return (
-      <MaybeSortableSelect
-        ref={setRef}
-        className={className}
-        classNamePrefix={classNamePrefix}
-        isMulti={isMulti}
-        isClearable={isClearable}
-        options={options}
-        value={value}
-        minMenuHeight={minMenuHeight}
-        maxMenuHeight={maxMenuHeight}
-        filterOption={
-          // filterOption may be NULL
-          filterOption !== undefined
-            ? filterOption
-            : createFilter({ ignoreAccents })
-        }
-        styles={{ ...DEFAULT_STYLES, ...stylesConfig } as SelectProps['styles']}
-        // merge default theme from `react-select`, default theme for Superset,
-        // and the theme from props.
-        theme={reactSelectTheme =>
-          merge(reactSelectTheme, defaultTheme(theme), themeConfig)
-        }
-        formatOptionLabel={formatOptionLabel}
-        getOptionLabel={getOptionLabel}
-        getOptionValue={getOptionValue}
-        components={components}
-        {...restProps}
-      />
+      <div ref={menuPortalTargetRef}>
```

Comment:
I've tried to test it and I can't find places where exact benefit of referring `menuPortalTarget` to the div. Maybe you could provide some gif or screens where it helps? Otherwise maybe `menuPortalTarget={document.body}` would be enough?

Reply:
Hi. @kkucharc thanks for your feedback. The `menuPortalTarget` to `document.body` will create the issue #12388 as the click event will be targeted outside of the Popover container. This is actually one of the main reasons why I am implementing this change. However, I think we can make this optional and just use it as a fallback, while managing all the other cases individually with ad hoc solutions where possible. I am looking into that.